Sources/SentryCrash/Recording/SentryCrashC.c

```diff
@@ -106,6 +107,17 @@ onCrash(struct SentryCrash_MonitorContext *monitorContext)
         strncpy(g_lastCrashReportFilePath, crashReportFilePath, sizeof(g_lastCrashReportFilePath));
         sentrycrashreport_writeStandardReport(monitorContext, crashReportFilePath);
     }
+
+    // Report is saved to disk, now we try to take screenshots.
+    // Depending on the state of the crash this may not work
+    // because we gonna call into asyng code.
+    if (g_saveScreenShot) {
+        unsigned long pathLen = strlen(g_lastCrashReportFilePath);
+        char crashScreenshotsPath[pathLen];
+        strcpy(crashScreenshotsPath, g_lastCrashReportFilePath);
+        crashScreenshotsPath[pathLen - 5] = 0; // remove extension to use as a directory
```

Comment:
`m`: How do we know how long the extension is? Instead of using `5` can we point to the place where the length of the extension is defined? This could break easily.

Reply:
Changing approach all together here.